src/solvers/dgsem_p4est/dg_3d_parabolic.jl

```diff
@@ -0,0 +1,697 @@
+                                               kk, element)
+                end
+            end
+
+            # now that the reference coordinate gradients are computed, transform them node-by-node to physical gradients
+            # using the contravariant vectors
+            for k in eachnode(dg), j in eachnode(dg), i in eachnode(dg)
+                Ja11, Ja12, Ja13 = get_contravariant_vector(1, contravariant_vectors, i, j,
+                                                            k,
+                                                            element)
+                Ja21, Ja22, Ja23 = get_contravariant_vector(2, contravariant_vectors, i, j,
+                                                            k,
+                                                            element)
+                Ja31, Ja32, Ja33 = get_contravariant_vector(3, contravariant_vectors, i, j,
+                                                            k,
+                                                            element)
+
+                gradients_reference_1 = get_node_vars(gradients_x, equations_parabolic, dg,
+                                                      i, j, k, element)
+                gradients_reference_2 = get_node_vars(gradients_y, equations_parabolic, dg,
+                                                      i, j, k, element)
+                gradients_reference_3 = get_node_vars(gradients_z, equations_parabolic, dg,
+                                                      i, j, k, element)
+
+                # note that the contravariant vectors are transposed compared with computations of flux
+                # divergences in `calc_volume_integral!`. See
+                # https://github.com/trixi-framework/Trixi.jl/pull/1490#discussion_r1213345190
+                # for a more detailed discussion.
+                gradient_x_node = Ja11 * gradients_reference_1 +
+                                  Ja21 * gradients_reference_2 +
+                                  Ja31 * gradients_reference_3
+                gradient_y_node = Ja12 * gradients_reference_1 +
+                                  Ja22 * gradients_reference_2 +
+                                  Ja32 * gradients_reference_3
+                gradient_z_node = Ja13 * gradients_reference_1 +
+                                  Ja23 * gradients_reference_2 +
+                                  Ja33 * gradients_reference_3
+
+                set_node_vars!(gradients_x, gradient_x_node, equations_parabolic, dg, i, j,
+                               k,
+                               element)
+                set_node_vars!(gradients_y, gradient_y_node, equations_parabolic, dg, i, j,
+                               k,
+                               element)
+                set_node_vars!(gradients_z, gradient_z_node, equations_parabolic, dg, i, j,
+                               k,
+                               element)
```

Comment:
```suggestion
                set_node_vars!(gradients_x, gradient_x_node, equations_parabolic, dg,
                               i, j, k, element)
                set_node_vars!(gradients_y, gradient_y_node, equations_parabolic, dg,
                               i, j, k, element)
                set_node_vars!(gradients_z, gradient_z_node, equations_parabolic, dg,
                               i, j, k, element)
```

Reply:
Will this be undone by JuliaFormatter? Do we need to put hand-formatted blocks?

Reply:
I fixed the issue and JuliaFormatter did not undo the changes.